jerry-core/jerry.c

```diff
@@ -651,893 +702,885 @@ jerry_create_error (jerry_error_t error_type, /**< type of error */
 {
   jerry_assert_api_available ();
 
-  return ecma_builtin_get (ECMA_BUILTIN_ID_GLOBAL);
-} /* jerry_get_global */
+  if (!ecma_is_value_object (obj_val))
+  {
+    return jerry_create_type_error ();
+  }
+
+  ecma_string_t *str_idx_p = ecma_new_ecma_string_from_uint32 ((uint32_t) index);
+  ecma_value_t ret_value = ecma_op_object_get (ecma_get_object_from_value (obj_val), str_idx_p);
+  ecma_deref_ecma_string (str_idx_p);
+
+  return ret_value;
+} /* jerry_get_property_by_index */
 
 /**
- * Perform eval
+ * Initialize property descriptor
+ */
+void
+jerry_init_property_descriptor (jerry_property_descriptor_t *prop_desc_p) /**< property descriptor */
+{
+  prop_desc_p->is_value_defined = false;
+  prop_desc_p->value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
+  prop_desc_p->is_writable_defined = false;
+  prop_desc_p->is_writable = false;
+  prop_desc_p->is_enumerable_defined = false;
+  prop_desc_p->is_enumerable = false;
+  prop_desc_p->is_configurable_defined = false;
+  prop_desc_p->is_configurable = false;
+  prop_desc_p->is_get_defined = false;
+  prop_desc_p->getter = ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
+  prop_desc_p->is_set_defined = false;
+  prop_desc_p->setter = ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
+} /* jerry_init_property_descriptor */
+
+/**
+ * Define a property to the specified object with the given name.
  *
  * Note:
- *      If current code is executed on top of interpreter, using is_direct argument,
- *      caller can enable direct eval mode that is equivalent to calling eval from
- *      within of current JS execution context.
+ *      returned value must be freed with jerry_release_value, when it is no longer needed.
  *
- * @return completion status
+ * @return true  - if success
+ *         thrown error - otherwise
  */
-jerry_completion_code_t
-jerry_eval (const jerry_char_t *source_p, /**< source code */
-            size_t source_size, /**< length of source code */
-            bool is_direct, /**< perform eval invocation in direct mode */
-            bool is_strict, /**< perform eval as it is called from strict mode code */
-            jerry_value_t *retval_p) /**< [out] returned value */
+jerry_value_t
+jerry_define_own_property (const jerry_value_t obj_val, /**< object value */
+                           const jerry_value_t prop_name_val, /**< property name (string value) */
+                           const jerry_property_descriptor_t *prop_desc_p) /**< property descriptor */
 {
   jerry_assert_api_available ();
 
-  jerry_completion_code_t status;
+  if (ECMA_IS_VALUE_ERROR (obj_val)
+      || ECMA_IS_VALUE_ERROR (prop_name_val))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG (error_value_msg_p));
+  }
+  else if (!ecma_is_value_object (obj_val)
+           || !ecma_is_value_string (prop_name_val))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG (wrong_args_msg_p));
+  }
```

Comment:
Please also check that the property descriptor is not both a data nd accessor property. That is:

    if (((writable_defined && writable) || has_value) && (has_getter || has_setter)) -> ERROR